File: src/main/java/repository/AccountRepo.java
package repository;

import domain.Account;
import domain.Artist;
import jdbcUtils.JdbcUtils;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.Properties;

public class AccountRepo implements AccountRepoInterface{
    private final JdbcUtils dbUtils;

    private static final Logger logger= LogManager.getLogger();

    public AccountRepo(Properties props) {
        logger.info("Initializing AccountRepo with properties: {}",props);
        dbUtils=new JdbcUtils(props);
    }

    @Override
    public Account getOne(String id) {
        logger.traceEntry();
        Connection con=dbUtils.getConnection();
        Account account=null;
        try(PreparedStatement preStmt=con.prepareStatement("select * from account where username=?")){
            preStmt.setString(1,id);
            try(ResultSet result=preStmt.executeQuery()){
                while(result.next()){
                    Long i=result.getLong("idUser");
                    String username=result.getString("username");
                    String password=result.getString("password");
                    account=new Account(username,i,password);
                }
            }
        }catch (SQLException ex){
            logger.error(ex);
            System.err.println("Error DB"+ex);
        }
        logger.traceExit(account);
        return account;
    }

    @Override
    public Iterable<Account> getAll() {
        return null;
    }

    @Override
    public Account add(Account entity) {
        logger.traceEntry("saving task {}",entity);
        Connection con=dbUtils.getConnection();
        try(PreparedStatement preStmt=con.prepareStatement("insert into account (username,idUser,password) values(?,?,?)")){
            preStmt.setString(1,entity.getUsername());
            preStmt.setLong(2,entity.getIdUser());
            preStmt.setString(2,entity.getPassword());
            int result=preStmt.executeUpdate();
            logger.trace("Saved {} instances",result);
        }catch (SQLException ex){
            logger.error(ex);
            System.err.println("Error DB"+ex);
        }
        logger.traceExit();
        return null;
    }

    @Override
    public Account delete(String s) {
        return null;
    }

    @Override
    public Account update(Account entity) {
        return null;
    }
}

File: repository/AccountRepo.cs
using LabMPP.domain;
using LabMPP.domain.Validators;
using LabMPP.JdbcUtils;
using log4net;
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.SqlClient;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LabMPP.repository
{
    class AccountRepo : AccountRepoInterface
    {
        SqlConnection connection = DBCUtils.GetConnection();
        private static readonly ILog log = LogManager.GetLogger("AccountRepo");
        public Account add(Account entity)
        {
            log.InfoFormat("entering add with value {0}", entity);
            SqlCommand sql = new SqlCommand("insert into account (username,password,idUser) values(@username,@password,@idUser)", connection);

            sql.Parameters.Add("@username", SqlDbType.VarChar).Value = entity.getUsername();
            sql.Parameters.Add("@password", SqlDbType.VarChar).Value = entity.password;
            sql.Parameters.Add("@idUser", SqlDbType.Int).Value = entity.idUser;

            connection.Open();

            var result = sql.ExecuteNonQuery();

            connection.Close();
            log.InfoFormat("exiting add with value {0}", result);
            if (result == 0)
                return entity;
            return null;
        }

        public Account delete(string id)
        {
            throw new NotImplementedException();
        }

        public IEnumerable<Account> getAll()
        {
            throw new NotImplementedException();
        }

        public Account getOne(string id)
        {
            log.InfoFormat("entering getOne with value {0}", id);
            SqlCommand sql = new SqlCommand("select * from account where username=@id", connection);

            sql.Parameters.Add("@id", SqlDbType.VarChar).Value = id;
            Account account = null;
            connection.Open();
            using (SqlDataReader reader = sql.ExecuteReader())
            {
                while (reader.Read())
                {
                    string username = reader["username"].ToString();
                    string password = reader["password"].ToString();
                    int i = int.Parse(reader["idUser"].ToString());
                    account = new Account(username, i, password);
                }
                connection.Close();
            }
            log.InfoFormat("exiting getOne with value {0}", account);
            return account;
        }

        public Account update(Account entity)
        {
            throw new NotImplementedException();
        }
    }
}
